internal/engine/upper_test.go

```diff
@@ -1268,3 +1347,11 @@ func (f *testFixture) assertAllHUDUpdatesConsumed() {
 		f.T().Fatalf("Update not consumed: %+v", update)
 	}
 }
+
+func (f *testFixture) assertAllActionsConsumed() {
```

Comment:
hmmmm...i'm not sure this is a good idea? in theory, the test has no control over what actions are consumed? can you add some comments on what this is trying to solve?

Reply:
I added this while trying to debug my failing test, and thought it might be good to leave in, but you're right, it's worse than nothing.